Web/src/components/join-room/index.tsx

```diff
@@ -1,135 +1,35 @@
+const JoinRoom: React.FC<RouteComponentProps<JoinRoomMatchParams>> = ({
   match: {
     params: { id },
-  },
+  }, history,
 }) => {
-  const [gamePin, setGamePin] = useState(id ? id.substring(0, PIN_LENGTH) : '');
-  const [username, setUsername] = useState('');
-
-  const [permission, setPermission] = useState(MotionPermission.NOT_SET);
-  const [showPermissionDialog, setPermissionDialog] = useState(false);
-  const [joinRequested, setJoinRequested] = useState(false);
-
+  const [initialId] = useState(id);
   const comm = useContext(CommContext);
-  const {
-    room, error,
-  } = useCommHooks(comm);
-  const history = useHistory();
-
-  const getUserPermission = async function requestPermission() {
-    try {
-      const permissionResult = await (window.DeviceMotionEvent as any).requestPermission();
-      if (permissionResult === 'granted') {
-        setPermission(MotionPermission.GRANTED);
-      } else {
-        setPermission(MotionPermission.DENIED);
-      }
-    } catch (e) {
-      setPermission(MotionPermission.DENIED);
-    } finally {
-      setPermissionDialog(false);
-    }
-  };
-
-  const getPermissionAvailability = () => {
-    if (!window.DeviceMotionEvent) {
-      setPermission(MotionPermission.DENIED);
-      return;
-    }
-    if (
-      typeof (window.DeviceMotionEvent as any).requestPermission === 'function'
-    ) {
-      setPermissionDialog(true);
-    } else {
-      setPermission(MotionPermission.GRANTED);
-    }
-  };
-
-  const onJoinRoomFormSubmit = (newGamePin: string, newUsername: string) => {
-    if (joinRequested) return;
-    if (newGamePin.length !== PIN_LENGTH) return;
-    if (newUsername === '') return;
-
-    // Reset state
-    setGamePin(newGamePin);
-    setUsername(newUsername);
-    setJoinRequested(true);
-  };
-
-  useEffect(() => {
-    if (joinRequested && permission === MotionPermission.NOT_SET) {
-      getPermissionAvailability();
-    }
-    if (joinRequested && permission === MotionPermission.GRANTED) {
-      comm.joinRoom(gamePin, username);
-    }
-  }, [permission, joinRequested, gamePin, username, comm]);
-
-  useEffect(() => {
-    if (error !== null) setJoinRequested(false);
-  }, [error]);
+  const commHooks = useCommHooks(comm);
 
   useEffect(() => {
-    if (room === null) return;
-    history.push('/participant-room');
-  }, [room, history]);
+    if (id) history.replace('/join');
+  }, [history, id]);
 
   return (
-    <JoinRoomContainer>
-      <PindaHead />
-      <JoinRoomForm
-        submitJoinRoomForm={onJoinRoomFormSubmit}
-        initialId={id}
-        pinLength={PIN_LENGTH}
-        permission={permission}
-      />
-      <Modal
-        isVisible={showPermissionDialog}
-        title="Give Permissions?"
-        onConfirm={getUserPermission}
-        confirmationButtonText="Sure!"
-      >
-        Pinda requires some device permissions in order to play some games.
-      </Modal>
-    </JoinRoomContainer>
+    <>
+      {commHooks.room === null
```

Comment:
we need to handle the case when error is not null, for example when the game is already ongoing

Reply:
We can do this in a separate PR though. Do you want to create an issue for this?

Reply:
what's the recovery strategy for this case? will error be nullfieid once the room becomes `FINISHED`?

Reply:
oop just saw the comment. let's do it in a separate PR.